authorizer/annotation.go

```diff
@@ -0,0 +1,171 @@
+package authorizer
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+)
+
+var _ influxdb.AnnotationService = (*AnnotationService)(nil)
+
+// AnnotationService wraps an influxdb.AnnotationService and authorizes actions
+// against it appropriately.
+type AnnotationService struct {
+	s influxdb.AnnotationService
+}
+
+// NewAnnotationService constructs an instance of an authorizing check service
+func NewAnnotationService(s influxdb.AnnotationService) *AnnotationService {
+	return &AnnotationService{
+		s: s,
+	}
+}
+
+// CreateAnnotations checks to see if the authorizer on context has write access for annotations for the provided orgID
+func (s *AnnotationService) CreateAnnotations(ctx context.Context, orgID platform.ID, create []influxdb.AnnotationCreate) ([]influxdb.AnnotationEvent, error) {
+	if _, _, err := AuthorizeCreate(ctx, influxdb.AnnotationsResourceType, orgID); err != nil {
+		return nil, err
+	}
+
+	return s.s.CreateAnnotations(ctx, orgID, create)
+}
+
+// ListAnnotations checks to see if the authorizer on context has read access for annotations for the provided orgID
+func (s *AnnotationService) ListAnnotations(ctx context.Context, orgID platform.ID, filter influxdb.AnnotationListFilter) (influxdb.ReadAnnotations, error) {
+	if _, _, err := AuthorizeOrgReadResource(ctx, influxdb.AnnotationsResourceType, orgID); err != nil {
```

Comment:
For other resources (i.e. buckets) we instead run the underlying `List` operation, then filter the results based on auth. Should we do the same here for consistency?

Reply:
Yeah this is interesting. Fundamentally I think that if you request a list of a resource for an org that you don't have read access to, you should get a forbidden response rather than an empty list. Otherwise there's no way to tell if you are requesting a list for an org that doesn't have any resources, or if you just aren't authenticated right. But indeed that is not consistent with how other resources work.

Trying to leverage as much of the IDPE code as I could, there wouldn't immediately be a way to get the entire list and then filter it by auth like the other resources, since the `ReadStream` and `ReadAnnotation` structs don't have the `orgID` in them - although I've already had to make some changes to responses and structs elsewhere so it's not a total show stopper.

Beyond my own personal biases and laziness, I think having it like this is going to be more consistent with the cloud API, although it's hard to say exactly how that will work since it doesn't have multi-org yet. I know that as it is now, the annotations service on cloud would give a forbidden response if you do a GET request for an org you don't have read access to, even if that's not really even possible to do.

So that's my reasoning for setting it up this way, mostly leaning into the consistency with cloud aspect, although I feel like there might be a gap with how other resources types respond on OSS vs. cloud. Annotations (and notebooks) are at least behind the `/api/v2private` endpoint so maybe there's some room for them to behave differently than other resources on OSS. Let me know what you think!

Reply:
Is the annotations permission an all-or-nothing within a single org? Or can users be granted read-permission on some-but-not-all of the annotations? If fine-grained auth is possible then I think we _have_ to post-filter the list results. I do like an explicit 403 on bad auth + consistency with cloud, so we could leave this top-level check in place either way.

Reply:
There's not fine-grained auth currently but who knows what the future holds. So I kept the top-level check and added the secondary filter! Seems like this should be good to keep things consistent and set us up for success in the future if fine-grained auth of things like annotations becomes necessary.